jcl/src/java.base/share/classes/openj9/internal/foreign/abi/InternalDowncallHandler.java

```diff
@@ -107,18 +108,21 @@ private static final class HeapArgInfo {
 		private long[] offsets;
 		private int index;
 		private final int size;
+		private MemorySessionImpl[] sessions;
```

Comment:
Just realized, this can actually be an array list `List<MemorySessionImpl>` because we dont actually care about the position.We just need to keep track of all of them.